cmd/context/create.go

```diff
@@ -133,13 +137,15 @@ func (c *ContextCommand) UseContext(ctx context.Context, ctxOptions *ContextOpti
 			transformedCtx := okteto.K8sContextToOktetoUrl(ctx, ctxOptions.Context, ctxOptions.Namespace, c.K8sClientProvider)
 			if transformedCtx != ctxOptions.Context {
 				ctxOptions.Context = transformedCtx
+				ctxOptions.Namespace = ctxStore.Contexts[ctxOptions.Context].Namespace
 				ctxOptions.IsOkteto = true
 			}
 		}
 	}
 
 	if okCtx, ok := ctxStore.Contexts[ctxOptions.Context]; !ok {
 		ctxStore.Contexts[ctxOptions.Context] = &okteto.OktetoContext{Name: ctxOptions.Context}
+		// What should be the namespace here?
 		created = true
 	} else if ctxOptions.Token == "" {
 		// this is to avoid login with the browser again if we already have a valid token
```

Comment:
I'd add it here checking if the namespace is not already set in `ctxOptions` (it can be set by the user in a flag and we should be using that one instead/if it's not done it will prevent running `okteto namespace`) and inside the code that you have above (`ctxOptions.Namespace = ctxStore.Contexts[ctxOptions.Context].Namespace`)

Reply:
Should I place `ctxOptions.Namespace = ctxStore.Contexts[ctxOptions.Context].Namespace` outside the `else if`? Wouldn't `ctxOptions.Namespace` not be set if `ctxOptions.Token` isn't empty?